src/main/java/edu/kit/compiler/codegen/ExitCondition.java

```diff
@@ -76,38 +92,53 @@ public void setFalseBlock(BlockEntry block) {
         private final Operand.Source target;
         private final Operand.Source source;
 
-        private BasicBlocks.BlockEntry trueBlock;
-        private BasicBlocks.BlockEntry falseBlock;
+        private int trueBlockId = -1;
+        private int falseBlockId = -1;
```

Comment:
Why the switch to using integers here? I know using `null` for missing values is not great, I'm not sure using `-1` instead is much of an improvement.

Reply:
The intention behind this change is not about `null`, instead the reason is that not every block that could be referenced here has a `BlockEntry`. Specifically, new blocks that are created for critical edges by the `PhiResolver` don't get a new entry (but still have an integer label)

Reply:
OK, I am can live with this change. The only thing we really lose, is the ability to navigate from a `BlockEntry` to its control flow successors. Though to be fair, we weren't using that to begin with, so it shouldn't be an issue.